### PR TITLE
3089 Update reserve transactions fetcher

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

File: accounting/apidocs/source/includes/_reserve_transactions.md [hunk 1]
```diff
@@ -72,4 +72,4 @@ Params | Type | Required | Default | Description
 ------ | ---- | -------- | ------- | -----------
 from | integer | true | one hour from now | from time to get transactions 
 to | integer | true | now | to time to get transactions 
-type | string | true | all | include: "normal", "internal", "ERC20" 
\ No newline at end of file
+type | string | true | all | include: "normal", "internal", "erc20"
```
Comment (favadi) [inline accounting/apidocs/source/includes/_reserve_transactions.md]: I think we should keep it ERC20.

File: accounting/reserve-transaction-fetcher/storage/postgres/postgres.go [hunk 2]
```diff
@@ -365,3 +366,37 @@ WHERE address ILIKE $1`
...
+		tmp common.ERC20Transfer
+	)
+	const selectStmt = `SELECT data FROM %[1]s WHERE ((data->>'timestamp')>=$1::text AND (data->>'timestamp')<$2::text) AND
+	($3 OR (data->>'from'=$4 OR data->>'to'=$4)) AND
+	($5 OR data->>'contractAddress'=$6)`
+	query := fmt.Sprintf(selectStmt, s.tableNames.ERC20)
+	logger.Debugw("querying ERC20 transfers history...", "query", query)
+	walletFilter := blockchain.IsZeroAddress(wallet)
+	tokenFilter := blockchain.IsZeroAddress(token)
+	if err := s.db.Select(&dbResult, query, timeutil.TimeToTimestampMs(from), timeutil.TimeToTimestampMs(to), walletFilter, wallet.Hex(), tokenFilter, token.Hex()); err != nil {
```
Comment (favadi) [inline accounting/reserve-transaction-fetcher/storage/postgres/postgres.go]: Does this query use index? Please show the reuslt with explain analyze. I think we might need to add more indices.

File: accounting/apidocs/source/includes/_reserve_transactions.md (hunk 1 above)
Comment (halink0803) [reply accounting/apidocs/source/includes/_reserve_transactions.md]: as discussed, we will keep it erc20 as it is input.

File: accounting/reserve-transaction-fetcher/storage/postgres/postgres.go (hunk 2 above)
Comment (halink0803) [reply accounting/reserve-transaction-fetcher/storage/postgres/postgres.go]: Added index for "wallet address" and "token address".

Query: 
```sql
EXPLAIN ANALYZE SELECT data FROM rsv_tx_erc20 WHERE ((data->>'timestamp')>= '1555125430000'::text AND (data->>'timestamp')<'1555211830000'::text) AND (data->>'from'='0x63825c174ab367968EC60f061753D3bbD36A0D8F' OR data->>'to'='0x63825c174ab367968EC60f061753D3bbD36A0D8F') AND (data->>'contractAddress'='0x2260FAC5E5542a773Aa44fBCfeD
f7C193bc2C599');
```

Result:
```
"Index Scan using rsv_tx_erc20_time_idx on rsv_tx_erc20  (cost=0.15..8.26 rows=1 width=32) (actual time=0.036..0.036 rows=0 loops=1)"
"  Index Cond: (((data ->> 'timestamp'::text) >= '1555125430000'::text) AND ((data ->> 'timestamp'::text) < '1555211830000'::text) AND ((data ->> 'contractAddress'::text) = '0x2260FAC5E5542a773Aa44fBCfeD"
"f7C193bc2C599'::text))"
"  Filter: (((data ->> 'from'::text) = '0x63825c174ab367968EC60f061753D3bbD36A0D8F'::text) OR ((data ->> 'to'::text) = '0x63825c174ab367968EC60f061753D3bbD36A0D8F'::text))"
"Planning time: 2.927 ms"
"Execution time: 0.251 ms"
```

**Pull request merged**